detector/zoo/plugin.go

```diff
@@ -6,6 +6,10 @@ import (
 
 func init() {
 	detector.Register("zk://", detector.PluginFactory(func(spec string) (detector.Master, error) {
-		return NewMasterDetector(spec)
+		md, err := NewMasterDetector(spec)
+		if err == nil {
+			go md.client.connect()
+		}
```

Comment:
Good catch!! I'm not in love with this. I'd rather do something similar to the code in `Standalone.Detect`, for example:
```go
func (md *MasterDetector) Detect(...) {
   // md.bootstrap would be a sync.Once variable
   bootstrap.Do(func() { go md.client.connect() })
   //.. the rest of Detect is fine as-is...
}
```